Updating the import guide based on the new CLI command (#4726)
pulumi/docs: content/docs/guides/adopting/import.md

* You need to reference existing resources to use as inputs to new resources in Pulumi
* You need to adopt existing resources under management so they can be managed by Pulumi

For the first situation, consult [the user guide index]({{< relref "/docs/guides/adopting#coexistence" >}}). For the
second, let's now see how to adopt existing resources.

## Adopting Existing Resources

To adopt existing resources so that Pulumi is able to manage subsequent updates to them, Pulumi offers the

> [`import`]({{< relref "/docs/reference/cli/pulumi_import" >}}) cli command option. This option request that a resource

Reviewer: Do we want to _replace_ the resource-level import documentation? I thought we'd want to offer this _in addition_, as an alternative, as the other programmatic model still offers benefits around automating the imports, no?

@lukehoban, curious for your feedback aslo.

Reviewer 2: Yes - I would like to keep both. The resource level approach is still useful for many scenarios, and thus is still the definitive documentation for that capability. 

I would lead with `pulumi import` as that is a simpler starting point.

    opts=ResourceOptions(protect=True))
```

{{< /choosable >}}
{{< choosable language go >}}

```go
package main

import (
	"github.com/pulumi/pulumi-aws/sdk/v3/go/aws/s3"
	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		_, err := s3.NewBucket(ctx, "infra-logs", &s3.BucketArgs{
			Acl:          pulumi.String("private"),
			Bucket:       pulumi.String("company-infra-logs"),
			ForceDestroy: pulumi.Bool(false),
		}, pulumi.Protect(true))
		if err != nil {
			return err
		}
		return nil
	})
}
```

{{< /choosable >}}
{{< choosable language csharp >}}

```csharp
using Pulumi;
using Aws = Pulumi.Aws;

class MyStack : Stack
{
    public MyStack()
    {
        var demo = new Aws.S3.Bucket("infra-logs", new Aws.S3.BucketArgs
        {
            Acl = "private",
            Bucket = "company-infra-logs",
            ForceDestroy = false,
        }, new CustomResourceOptions
        {
            Protect = true,
        });
    }

}
```

{{< /choosable >}}
{{< /chooser >}}

> Your Pulumi stack must be configured correctly---e.g., in this case, the correct AWS region---otherwise the resource will not be found.

> After successfully importing a resource, you can run `pulumi up`, and all subsequent operations will behave as though Pulumi provisioned
> the resource from the outset. When Pulumi performs an import, the resource is added to the Pulumi [state](https://www.pulumi.com/docs/intro/concepts/state/#state),
> it is marked as a [protected](https://www.pulumi.com/docs/intro/concepts/programming-model/#protect) resource (by default),
> and it will emit the generated code that the user can add to their program before running a `pulumi up`. The resources are
> marked as protected to ensure that imported infrastructure is not accidentally deleted if the user forgets to include
> the code for the resource in their program before doing a deployment.

Reviewer 3: Some of the added text here was covered above, so I took a stab at summarizing existing information and keeping the new info in about the protected nature of these resources.

```suggestion
After successfully importing a resource and adding the generated code to your program, you can run `pulumi up` and all subsequent operations will behave as though Pulumi provisioned
the resource from the outset. The resource is added to the Pulumi [state](https://www.pulumi.com/docs/intro/concepts/state/#state),
and marked as a [protected](https://www.pulumi.com/docs/intro/concepts/programming-model/#protect) resource (by default) to ensure that imported infrastructure is not accidentally deleted if the user forgets to include the code for the resource in their program before doing a deployment.
```

### Bulk Import Operation

> Should a user need to import multiple resources to be managed by Pulumi, the CLI `import command` can handle this
> scenario for us. This is helpful when using the pulumi import command as part of scripting larger bulk imports of
> cloud resource:

Reviewer 3: ```suggestion
If you need to import multiple resources, the CLI `import` command can be used with a JSON file that contains references to existing cloud resources. Using a JSON file with the `import` command can be helpful as part of scripting large bulk imports of cloud resources.
```

error: Preview failed: importing sg-04aeda9a214730248: security group not found
```

After successfully importing a resource, you can delete the `import` statement, rerun `pulumi up`, and all subsequent operations will behave as though Pulumi provisioned the resource from the outset. Be careful, as this applies to `destroy` operations also.

#### Mismatched State

> An important part of importing resources via resource options is that the resulting Pulumi program, after the import is complete, will faithfully generate the same desired state as your existing infrastructure's actual state. After the import, of course, you may edit your program to generate and apply new desired states to update your infrastructure.

Reviewer 3: ```suggestion
An important part of importing resources using the `import` resource option is that the resulting Pulumi program, after the import is complete, will faithfully generate the same desired state as your existing infrastructure's actual state. After the import, you may edit your program to generate and apply new desired states to update your infrastructure.
```
